File: Phase-2/frontend/types/task.ts
/**
 * Task type definitions
 */

export enum Priority {
  LOW = "low",
  MEDIUM = "medium",
  HIGH = "high",
}

export enum Recurrence {
  NONE = "none",
  DAILY = "daily",
  WEEKLY = "weekly",
  MONTHLY = "monthly",
}

export interface Task {
  id: number
  user_id: number
  title: string
  description: string | null
  completed: boolean
  priority: Priority
  category: string | null
  due_date: string | null
  recurrence: Recurrence
  created_at: string
  updated_at: string
  is_overdue: boolean
  is_due_today: boolean
}

export interface TaskCreate {
  title: string
  description?: string | null
  priority?: Priority
  category?: string | null
  due_date?: string | null
  recurrence?: Recurrence
}

export interface TaskUpdate {
  title?: string
  description?: string | null
  completed?: boolean
  priority?: Priority
  category?: string | null
  due_date?: string | null
  recurrence?: Recurrence
}

File: Phase-2/backend/src/models/task.py
"""Task database model."""
from datetime import datetime
from enum import Enum
from typing import Optional

from sqlmodel import Field, SQLModel


class Priority(str, Enum):
    """Task priority levels."""

    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"


class Recurrence(str, Enum):
    """Task recurrence patterns."""

    NONE = "none"
    DAILY = "daily"
    WEEKLY = "weekly"
    MONTHLY = "monthly"


class Task(SQLModel, table=True):
    """Task model for todo items."""

    __tablename__ = "tasks"

    id: Optional[int] = Field(default=None, primary_key=True)
    user_id: str = Field(index=True, max_length=255)  # Better-Auth uses string IDs
    title: str = Field(max_length=255)
    description: Optional[str] = Field(default=None)
    completed: bool = Field(default=False)
    priority: Priority = Field(default=Priority.MEDIUM)
    category: Optional[str] = Field(default=None, max_length=100)
    due_date: Optional[datetime] = Field(default=None)
    recurrence: Recurrence = Field(default=Recurrence.NONE)
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)
